src/bin/purge_ttl.rs

```diff
@@ -69,32 +79,184 @@ pub fn statsd_from_env() -> Result<StatsdClient, Box<dyn Error>> {
         .build())
 }
 
+pub enum RequestType {
+    ReadOnly,
+    ReadWrite,
+    PartitionedDml,
+}
+
 fn prepare_request(
     client: &SpannerClient,
     session: &Session,
-) -> Result<ExecuteSqlRequest, Box<dyn Error>> {
+    request_type: RequestType,
+) -> Result<(ExecuteSqlRequest, Vec<u8>), Box<dyn Error>> {
     // Create a transaction
     let mut opt = TransactionOptions::new();
-    opt.set_partitioned_dml(TransactionOptions_PartitionedDml::new());
+    match request_type {
+        RequestType::ReadWrite => {
+            opt.set_read_write(TransactionOptions_ReadWrite::new());
+        }
+        RequestType::ReadOnly => {
+            opt.set_read_only(TransactionOptions_ReadOnly::new());
+        }
+        RequestType::PartitionedDml => {}
+    }
     let mut req = BeginTransactionRequest::new();
     req.set_session(session.get_name().to_owned());
     req.set_options(opt);
     let mut txn = client.begin_transaction(&req)?;
 
     let mut ts = TransactionSelector::new();
-    ts.set_id(txn.take_id());
+    let id = txn.take_id();
+    ts.set_id(id.clone());
 
     // Create an SQL request
     let mut req = ExecuteSqlRequest::new();
     req.set_session(session.get_name().to_string());
     req.set_transaction(ts);
 
-    Ok(req)
+    Ok((req, id))
+}
+
+fn commit_request(
+    client: &SpannerClient,
+    session: &Session,
+    txn: Vec<u8>,
+) -> Result<(), Box<dyn Error>> {
+    let mut req = CommitRequest::new();
+    req.set_session(session.get_name().to_owned());
+    req.set_transaction_id(txn);
+    client.commit(&req)?;
+    Ok(())
+}
+
+pub struct SyncResultSet {
+    result: ResultSet,
+}
+
+impl SyncResultSet {
+    pub fn stats(&self) -> Option<&ResultSetStats> {
+        self.result.stats.as_ref()
+    }
+}
+
+impl Iterator for SyncResultSet {
+    type Item = Vec<Value>;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        let rows = &mut self.result.rows;
+        if rows.is_empty() {
+            None
+        } else {
+            let row = rows.remove(0);
+            Some(row.get_values().to_vec())
+        }
+    }
+}
+
+fn delete_incremental(
+    client: &SpannerClient,
+    session: &Session,
+    table: String,
+    column: String,
+    batch_size: u64,
+) -> Result<(), Box<dyn Error>> {
+    let (mut req, _txn) = prepare_request(&client, &session, RequestType::ReadOnly)?;
+    let sql = format!("SELECT collection_id, fxa_kid, fxa_uid, {} FROM {} WHERE expiry < CURRENT_TIMESTAMP() LIMIT 1000000", column, table);
```

Comment:
nit: I'd prefer the order of columns here matches what we use elsewhere (in the schema and spanner impl), uid, kid, cid, id. For what it's worth I think it's helpful for any new eyes looking at this stuff (and probably long term maintenance -- didn't bother with this on the migration scripts because they're temporary)

Reply:
Sure. Done